androidApp/src/main/java/dev/johnoreilly/confetti/sessions/HomeViewModel.kt

```diff
@@ -0,0 +1,61 @@
+package dev.johnoreilly.confetti.sessions
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dev.johnoreilly.confetti.ConfettiRepository
+import dev.johnoreilly.confetti.sessions.navigation.SessionsKey
+import dev.johnoreilly.confetti.splash.SplashReadyStatus
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.WhileSubscribed
+import kotlinx.coroutines.flow.emitAll
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.stateIn
+import kotlin.time.Duration.Companion.seconds
+
+class HomeViewModel(
+    savedStateHandle: SavedStateHandle,
+    confettiRepository: ConfettiRepository,
+    splashReadyStatus: SplashReadyStatus,
+) : ViewModel() {
+    private val conferenceParam: String? =
+        SessionsKey.conferenceFromNavArgs(savedStateHandle)?.ifEmpty { null }
+
+    private val conferenceFlow: Flow<String> = flow {
+        if (conferenceParam != null) {
+            // Set the conference selected if we've deep linked into a specific conference
+            confettiRepository.setConference(conferenceParam)
```

Comment:
This is the part that makes sure that if we've deep linked into a specific conference we also select it in the repository.
This made sense to me so that if you deep link into one, then close the app, then open it again I'd suspect you'd want to see what you saw last time too, not what you pressed the time *before* the last time you had the app open.